### DISCUSSION TITLE
Monitoring the queue size

### DISCUSSION THREAD

**Discussion by dometec** [smallrye/smallrye-reactive-messaging#1758] [Q&A]
Hi Team!
I'm starting using this library to receive data from an MQTT and write it on a TimeSeries database with a little transformation. 
I want to monitor how many messages there are in the queue and are waiting to get acked:

```
@Incoming("data")
@Blocking(ordered = false)
public CompletionStage<Void> readFromMQTT(Message<byte[]> data) {
   // TODO transfor and write to database
   return confdocData.ack();
}
```

There is a way to get the count? Maybe on the /q/metric endpoint?

Thanks,
Domenico

**Comment by PyAntony** [comment]
There doesn't seem to be anything available for this. I have created a bunch of custom code to keep track of counts/throughput.
